src/mon/mon_types.h

```diff
@@ -736,4 +737,75 @@ struct ProgressEvent {
 };
 WRITE_CLASS_ENCODER(ProgressEvent)
 
+struct PoolAvailability {
+  std::string pool_name;
+  utime_t started_at;
+  uint64_t uptime;
+  utime_t last_uptime;
+  uint64_t downtime;
+  utime_t last_downtime;
+  uint64_t num_failures;
+  bool is_avail;
+
+  PoolAvailability()
+    : pool_name(""), started_at(ceph_clock_now()),
+      uptime(0), last_uptime(ceph_clock_now()),
+      downtime(0), last_downtime(ceph_clock_now()),
+      num_failures(0), is_avail(true)
```

Comment:
Let's initialize these inline where the members are declared where possible.  Avoids missing them when adding a new constructor.